packages/datadog-plugin-kafkajs/src/producer.js

```diff
@@ -1,13 +1,22 @@
 'use strict'
 
 const ProducerPlugin = require('../../dd-trace/src/plugins/producer')
+const { encodePathwayContext } = require('../../dd-trace/src/datastreams/pathway')
 
 class KafkajsProducerPlugin extends ProducerPlugin {
   static get id () { return 'kafkajs' }
   static get operation () { return 'produce' }
 
   start ({ topic, messages }) {
-    const span = this.startSpan({
+
+    let pathwayCtx
+    if (this.config.dsmEnabled) {
+      const dataStreamsContext = this.tracer
+        .setCheckpoint(['direction:out', `topic:${topic}`, 'type:kafka'])
+      pathwayCtx = encodePathwayContext(dataStreamsContext)
+    }
+    const span = this.startSpan('kafka.produce', {
+      service: this.config.service || `${this.tracer._service}-kafka`,
```

Comment:
Similar to the above, why are we changing the service name logic?

Reply:
Another rebase issue